### PR TITLE
Journeyman/learned concept

### PR DESCRIPTION
**Pull request by journeyman-msft**
Using a learned selector for quanser qube.
image

Please confirm:
1. Training for the selector is similar (NPIL of 1mil was found to be needed) 
image

2. Export brain work with visualizer using `python main.py --render --test-exported <PORT>`

3. automated test
```
pytest -s \
    --acr_name <ACR_NAME> \
    --brain_name <BRAIN_NAME> \
    --port <PORT> \
    --render False \
```

### CODE REVIEW

File: Python/samples/quanser-qube/tests/test_multi_concept.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
+"""
+Pytest for confirming multi concept brain functions with learned selector,
+one learned concept, and one programmed concept. Assessment with multi-concept
```
Comment (shnayder) [inline Python/samples/quanser-qube/tests/test_multi_concept.py:3]: what do you mean by "assessment with multi-concept is not available"? It should work fine to assess the selector concept, though testing with the exported brain seems perfectly fine too. Suggest just removing this sentence.

Comment (journeyman-msft) [reply Python/samples/quanser-qube/tests/test_multi_concept.py:3]: I learned this on Friday. I misunderstood M5 was assessment for programmed concepts (not selectors). I can fix

**Comment by shnayder** [comment]
I'm just about to try training this myself, but based on the code this all looks good.

**Comment by akzaidi** [comment]
image

tried this yesterday and worked great. exported worked like a charm too. might not need as long as an NPIL, selector finished training after 1.2M and swingup after 500k.

exported results:

output

**Comment by journeyman-msft** [comment]
> image
> 
> tried this yesterday and worked great. exported worked like a charm too. might not need as long as an NPIL, selector finished training after 1.2M and swingup after 500k.
> 
> exported results:
> 
> output

That's great to hear! Perhaps i'll lower the NPIL to 500k and see if it still works. Thanks for the gif too, i'll absorb that into the README.

**Comment by journeyman-msft** [comment]
Interestingly, 500k was not enough
image

**Comment by shnayder** [comment]
My test also worked! Training the selector only took ~800K iterations this time:
image

@journeyman-msft  I ran into a bunch of python issues trying to get the sim to run locally. Traced it to pinned versions in requirements.txt — do you really need those exact versions? In particular scipy wouldn't build on my version of OS X until I removed the pin. And wouldn't you want latest bonsai-cli too?

**Comment by shnayder** [comment]
Now I want to know what the decision boundary the selector learned looks like. Once the new exported brain APIs report which concept was chosen, we can make some plots  :)

**Comment by shnayder** [comment]
@journeyman-msft two small notes: 
1. can you add the reference frame for the two state angles to the comments? (i.e. is 0 radians straight up or down?). 
2. I just saw Mike trying to create a brain from sim for qube and then saw errors about missing state vars—does the interface file promise some target values that aren't actually sent by the sim? (I haven't looked at the code to confirm—might have been some other issue)

**Comment by journeyman-msft** [comment]
- removed target_ items in interface.json
- increased range of alpha for in interface.json
- added img for reference frame

image

**Pull request merged**